src/assets/wise5/services/studentPeerGroupService.ts

```diff
@@ -0,0 +1,149 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { PeerGroupStudentData } from '../../../app/domain/peerGroupStudentData';
+import { PeerGroup } from '../components/peerChat/PeerGroup';
+import { PeerGroupMember } from '../components/peerChat/PeerGroupMember';
+import { AnnotationService } from './annotationService';
+import { ConfigService } from './configService';
+import { PeerGroupService } from './peerGroupService';
+import { StudentDataService } from './studentDataService';
+import { ProjectService } from './projectService';
+
+@Injectable()
+export class StudentPeerGroupService extends PeerGroupService {
+  static readonly PREVIEW_PEER_GROUP_ID = 1;
+
+  constructor(
+    private annotationService: AnnotationService,
+    protected configService: ConfigService,
+    protected http: HttpClient,
+    private projectService: ProjectService,
+    private studentDataService: StudentDataService
+  ) {
+    super(configService, http);
+  }
+
+  retrievePeerGroup(
+    peerGroupingTag: string,
+    workgroupId = this.configService.getWorkgroupId()
+  ): Observable<PeerGroup> {
+    if (
+      this.configService.isPreview() ||
+      this.configService.isAuthoring() ||
+      this.configService.isSignedInUserATeacher()
+    ) {
+      return of(this.getPreviewPeerGroup());
+    }
+    return super.retrievePeerGroup(peerGroupingTag, workgroupId);
+  }
+
+  private getPreviewPeerGroup(): PeerGroup {
+    let workgroupId = this.configService.getWorkgroupId();
+    let periodId = this.configService.getPeriodId();
+    if (this.configService.isAuthoring()) {
+      workgroupId = 1;
+      periodId = 1;
+    }
```

Comment:
Could reactor so we only need to call `this.configService.getWorkgroupId()` and `this.configService.getPeriodId()` if we're not in authoring mode.

```suggestion
    let workgroupId = 1;
    let periodId = 1;
    if (!this.configService.isAuthoring()) {
      workgroupId = this.configService.getWorkgroupId();
      periodId = this.configService.getPeriodId();
    }
```